docs/decisions/0002-limiting-stream-length.rst

```diff
@@ -0,0 +1,68 @@
+0002 Limiting stream length
+###########################
+
+Status
+******
+
+**Provisional**
+
+Context
+*******
+
+The redis streams are an append-only data structure that stores entries
+forever. All the triggered events will be stored in the stream and eventually
+consume all the available memory. However storing these events forever is
+not required so we can set a limit to the number entries that can be stored,
+removing older entries whenever the limit is crossed.
+
+Decision
+********
+
+There are two options to cap these streams to a specific length.
+
+Running `XTRIM`_ command regularly by creating custom django command. It
+supports two strategies to evict older entries.
+
+.. _XTRIM: https://redis.io/commands/xtrim/
+
+
+* MAXLEN: Evicts entries as long as the stream's length exceeds the specified
+  threshold, where threshold is a positive integer.
+* MINID: Evicts entries with IDs lower than threshold, where threshold is a
+  stream ID.
+
+
+Another option which only provides single strategy i.e. ``MAXLEN`` but
+simpler in terms of implementation as well as maintenance is including maximum
+length of stream while adding items to the stream.
+
+.. code-block::
+
+   stream.add(data, id='*', maxlen=None, approximate=True)
+
+We can set maxlen value via django settings with a default value of 1000 (Bear
+in mind that this estimate is only a rough one). Setting ``approximate=True``
+is required as trimming to exact length can be expensive operation. Streams are
```

Comment:
I think the detail on how streams are store in memory is probably more than we need here, and we can just say that it will prevent memory issues by capping the max number of events, and that using "approximate" is faster, and let the link explain the rest if folks care for the technical details.

One thing we may want to note is that messages on the stream are persisted to the AOF and replicated, so while they may get removed from the stream they're not unrecoverable in the case of an issue. Same thing with that one, just linking out to the docs for the details should do.

Reply:
Done.